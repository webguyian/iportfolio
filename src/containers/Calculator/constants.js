export const add = '+';
export const allClear = 'AC';
export const backspace = 'Backspace';
export const clear = 'C';
export const decimal = '.';
export const divide = '/';
export const divideAlt = '\u00f7';
export const enter = 'Enter';
export const equals = '=';
export const multiply = '*';
export const multiplyAlt = '\u00d7';
export const percent = '%';
export const positiveNegative = '+/-';
export const subtract = '-';
export const zero = '0';

export const controls = {
  add,
  allClear,
  clear,
  decimal,
  divide,
  divideAlt,
  equals,
  multiply,
  multiplyAlt,
  percent,
  positiveNegative,
  subtract
};

export const allOperators = [
  add,
  allClear,
  clear,
  decimal,
  divide,
  divideAlt,
  equals,
  multiply,
  multiplyAlt,
  percent,
  positiveNegative,
  subtract
];

export const operators = [
  add,
  allClear,
  clear,
  divide,
  divideAlt,
  multiply,
  multiplyAlt,
  positiveNegative,
  subtract
];

export const controlLabels = {
  [allClear]: 'All clear',
  [clear]: 'Clear',
  [positiveNegative]: 'Positive/negative',
  [percent]: 'Percent',
  [multiplyAlt]: 'Multiply',
  [divideAlt]: 'Divide',
  [add]: 'Add',
  [subtract]: 'Subtract',
  [equals]: 'Equals',
  [decimal]: 'Decimal'
};

export const numbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];

export const keys = {
  backspace,
  enter
};

export const operatorKeys = [backspace, enter]
  .concat(...allOperators)
  .concat(numbers);
