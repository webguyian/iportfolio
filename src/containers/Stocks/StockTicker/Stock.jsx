import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { formatPrice, getPercentage } from 'modules/stocks/helpers';

import Button from 'components/Button/Button';
import Text from 'components/Text/Text';
import StockChart from 'containers/Stocks/StockChart/StockChart';

const Stock = props => {
  const { onClick, stock } = props;
  const { price, previousPrice, symbol } = stock;
  const baseClass = 'stock-tick';
  const differenceClass = `${baseClass}-difference`;
  const modifierClass =
    price > previousPrice
      ? `${differenceClass}--up`
      : `${differenceClass}--down`;

  return (
    <li className={baseClass}>
      <Button className={`${baseClass}-btn`} onClick={onClick}>
        <div className={`${baseClass}-content`}>
          <Text className={`${baseClass}-symbol`} modifier="bold">
            {symbol}
          </Text>
          <Text className={`${baseClass}-price`} modifier="bold">
            {formatPrice(price)}
          </Text>
          <Text className={classNames(differenceClass, modifierClass)}>
            {getPercentage(price, previousPrice)}
          </Text>
        </div>
        <StockChart
          hideControls
          stock={stock}
          mouseMoveEvent={false}
          panEvent={false}
          zoomEvent={false}
        />
      </Button>
    </li>
  );
};

Stock.propTypes = {
  description: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  stock: PropTypes.object.isRequired
};

export default Stock;
