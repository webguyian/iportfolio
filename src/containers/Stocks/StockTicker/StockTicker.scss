.stock-ticker {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 70px;
  padding-left: 10%;
  background-color: $black-darker;

  &::before,
  &::after {
    z-index: 5;
    position: absolute;
    top: 0;
    left: 0;
    width: $gutter;
    height: 70px;
    background-color: $black-darker;
    content: "";
  }

  &::after {
    left: 360px;
  }

  &-list {
    @include unstyled-list;
    display: flex;
    flex-flow: row nowrap;
    margin: 0 $gutter;
    animation: marquee 30s linear infinite;
  }

  .stock-tick {
    &-btn {
      position: relative;
      width: 100%;
      min-width: 200px;
      padding: $gutter-s;
      color: $white;
      text-align: left;
    }

    &-content {
      display: flex;
      flex-flow: column wrap;
    }

    .stock-chart {
      position: absolute;
      top: -60px;
      left: -30px;
      transform: scale(0.4);
    }

    &-symbol,
    &-price {
      font-size: $font-size-s;
    }

    &-price {
      margin-bottom: 4px;
    }

    &-difference {
      &--down {
        color: $red;
      }

      &--up {
        color: $green-light;
      }
    }
  }
}

@keyframes marquee {
  0% {
    transform: translateX(0);
  }

  100% {
    transform: translateX(-80%);
  }
}
