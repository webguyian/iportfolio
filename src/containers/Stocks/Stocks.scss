.stocks-app {
  @include stretch-column-nowrap;
  position: relative;
  padding: $gutter;
  transition: transform 0.3s ease;

  &--with-search {
    padding-top: 70px;
    transform: translateY(-70px);

    .stocks-app-header {
      display: none;
    }

    .stocks-app-list {
      max-height: none;
    }

    .ui-search-field {
      margin: $gutter-s 4px;

      > .ui-btn--anchor {
        display: block; // Show cancel button
      }
    }
  }

  &--with-detail {
    padding-top: $gutter;
    transform: none;

    > .stock-news {
      transform: translateY(110%);
    }
  }

  &-header {
    display: flex;
    align-items: flex-start;
    justify-content: space-between;
    padding: 0 $gutter-s;
  }

  &-title,
  &-date {
    font-size: $font-size-l;
  }

  &-date {
    color: $gray;
  }

  &-edit-btn {
    @include type-body;
    padding: 4px $gutter;
    color: $blue-light;
  }

  &-list {
    @include unstyled-list;
    max-height: 600px;
    overflow-y: auto;
  }
}
