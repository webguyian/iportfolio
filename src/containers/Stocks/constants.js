export const initialStocks = [
  { symbol: 'AAPL', displaySymbol: 'AAPL', description: 'APPLE INC' },
  { symbol: 'AMZN', displaySymbol: 'AMZN', description: 'AMAZON.COM INC' },
  { symbol: 'FB', displaySymbol: 'FB', description: 'FACEBOOK INC-CLASS A' },
  { symbol: 'GOOG', displaySymbol: 'GOOG', description: 'ALPHABET INC-CL C' },
  { symbol: 'MSFT', displaySymbol: 'MSFT', description: 'MICROSOFT CORP' }
];

export const API_CANDLESTICK = 'https://finnhub.io/api/v1/stock/candle';

export const API_NEWS = 'https://finnhub.io/api/v1/news';

export const API_QUOTE = 'https://finnhub.io/api/v1/quote';

export const API_TOKEN = 'bojsah7rh5rcji5m4udg';

export const API_EXCHANGE = `https://finnhub.io/api/v1/stock/symbol?exchange=US&token=${API_TOKEN}`;
