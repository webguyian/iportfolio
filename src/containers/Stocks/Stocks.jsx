import React, { useState } from 'react';
import classNames from 'classnames';

import { initialStocks } from './constants';
import { useSearch, useStockSearch, useStocks, useStockDetail } from './hooks';

import Button from 'components/Button/Button';
import DateTime from 'components/DateTime/DateTime';
import Text from 'components/Text/Text';

import Stock from 'containers/Stocks/Stock/Stock';
import StockDetail from 'containers/Stocks/Stock/StockDetail';
import StockSearch from 'containers/Stocks/StockSearch/StockSearch';
import StockNews from 'containers/Stocks/StockNews/StockNews';

const Stocks = () => {
  const [stocks, setStocks, allStocks] = useStocks(initialStocks);
  const [searchTerm, searchHandlers, hasSearch] = useSearch();
  const filteredStocks = useStockSearch(allStocks, searchTerm);
  const [activeStock, setActiveStock] = useStockDetail();
  const [isEdit, setEdit] = useState(false);
  const visibleStocks = searchTerm.length > 1 ? filteredStocks : stocks;
  const baseClass = 'stocks-app';
  const withDetailClass =
    (activeStock || isEdit) && `${baseClass}--with-detail`;
  const withSearchClass = hasSearch && `${baseClass}--with-search`;
  const actionLabel = isEdit ? 'Done' : 'Edit';
  const handleAddStock = () => {
    setStocks(stocks.concat(activeStock));
    setActiveStock(null);
    searchHandlers.onCancel();
  };
  const deleteStock = currentStock => {
    setStocks(stocks.filter(stock => stock !== currentStock));
  };
  const handleEdit = () => {
    setEdit(edit => !edit);
  };

  return (
    <div className={classNames(baseClass, withDetailClass, withSearchClass)}>
      <header className={`${baseClass}-header`}>
        <div className={`${baseClass}-header-content`}>
          <Text className={`${baseClass}-title`} element="h1" type="display">
            Stocks
          </Text>
          <Text className={`${baseClass}-date`} element="h2" type="display">
            <DateTime format="MMMM d" />
          </Text>
        </div>
        <Button className="ui-btn--anchor" onClick={handleEdit}>
          {actionLabel}
        </Button>
      </header>
      <StockSearch {...searchHandlers} />
      <ul className={`${baseClass}-list`}>
        {visibleStocks.map(stock => (
          <Stock
            key={stock.displaySymbol}
            {...stock}
            onClick={setActiveStock.bind(null, stock)}
            onDelete={deleteStock.bind(null, stock)}
            hidePrice={isEdit}
          />
        ))}
      </ul>
      <StockDetail
        stock={activeStock}
        stocks={stocks}
        onAdd={handleAddStock}
        onClick={setActiveStock}
        onClose={setActiveStock.bind(null, null)}
      />
      <StockNews draggable />
    </div>
  );
};

export default Stocks;
