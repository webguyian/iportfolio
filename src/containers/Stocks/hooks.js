import { useEffect, useState } from 'react';

import { useStorageCache } from 'hooks';
import { API_EXCHANGE, API_TOKEN } from 'containers/Stocks/constants';
import {
  getCandlestickEndpoint,
  getCandlestickEndpoints,
  getStockEndpoint,
  getStockNewsEndpoint,
  isExpired,
  isExpiredNews,
  updateChartData,
  rehydrateChartData
} from 'containers/Stocks/helpers';

export const useFetch = endpoint => {
  const [data, setData] = useState(null);
  const fetchData = async () => {
    const response = await fetch(endpoint);
    const result = await response.json();

    setData(result);
  };

  useEffect(() => {
    if (!endpoint) {
      return;
    }

    fetchData();
  }, [endpoint]);

  return data;
};

export const useFetchAll = urls => {
  const [responses, setResponses] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const results = await Promise.all(
        urls.map(async url => {
          const response = await fetch(url);

          return response.json();
        })
      );

      setResponses(results);
    };

    if (urls.length) {
      fetchData();
    }
  }, [urls]);

  return responses;
};

export const useSearch = () => {
  const [value, setValue] = useState('');
  const [searchTerm, setSearchTerm] = useState('');
  const [searchFocus, setSearchFocus] = useState(false);
  const hasSearch = Boolean(searchFocus || searchTerm.length);

  const handleCancel = () => {
    setValue('');
    setSearchTerm('');
    setSearchFocus(false);
  };

  const handleChange = event => {
    setValue(event.target.value);
  };

  const handleFocus = () => {
    setSearchFocus(focus => !focus);
  };

  const searchHandlers = {
    onBlur: handleFocus,
    onCancel: handleCancel,
    onChange: handleChange,
    onFocus: handleFocus,
    value
  };

  useEffect(() => {
    if (value) {
      setSearchTerm(value.toUpperCase().trim());
    } else {
      setSearchTerm('');
    }
  }, [value]);

  return [searchTerm, searchHandlers, hasSearch];
};

export const useStockSearch = (allStocks, searchTerm) => {
  const [filteredStocks, setFilteredStocks] = useState(allStocks);

  useEffect(() => {
    if (searchTerm.length > 1) {
      const filtered = allStocks.filter(stock => {
        return (
          stock.symbol.indexOf(searchTerm) === 0 ||
          stock.description.indexOf(searchTerm) === 0
        );
      });

      setFilteredStocks(filtered);
    } else if (!searchTerm) {
      setFilteredStocks(allStocks);
    }
  }, [searchTerm]);

  return filteredStocks;
};

export const useStocks = initialStocks => {
  const [endpoints, setEndpoints] = useState([]);
  const [stocks, setStocks] = useState([]);
  const [allStocks, setAllStocks] = useState([]);
  const [activeStocks, setActiveStocks] = useState([]);
  const exchangeResponse = useFetch(API_EXCHANGE);
  const response = useFetchAll(endpoints);
  const cache = useStorageCache(
    'stocks',
    { stocks },
    stockData => !stockData.stocks.length
  );

  useEffect(() => {
    const cachedStocks = cache && cache.stocks;

    if (cachedStocks && cachedStocks.length) {
      setStocks(cachedStocks);
      setActiveStocks(cachedStocks);
    } else {
      setStocks(initialStocks);
      setActiveStocks(initialStocks);
    }
  }, []);

  useEffect(() => {
    if (!exchangeResponse) {
      // Exit early if no response
      return;
    }

    const symbols = activeStocks.map(stock => stock.symbol);
    const currentStocks = exchangeResponse.filter(stock =>
      symbols.includes(stock.symbol)
    );

    setEndpoints(currentStocks.map(getStockEndpoint));
    setAllStocks(exchangeResponse);
  }, [exchangeResponse]);

  useEffect(() => {
    if (activeStocks.length) {
      setEndpoints(activeStocks.map(getStockEndpoint));
    }
  }, [activeStocks]);

  useEffect(() => {
    if (!response.length || !activeStocks.length) {
      return;
    }

    const updatedStocks = response.map((result, index) => {
      const stock = activeStocks[index];

      return {
        ...result,
        ...stock,
        price: result.c,
        previousPrice: result.pc
      };
    });

    setStocks(updatedStocks);
  }, [response, activeStocks]);

  return [stocks, setStocks, allStocks];
};

export const useStockDetail = () => {
  const [endpoint, setEndpoint] = useState('');
  const [activeStock, setActiveStock] = useState(null);
  const [stock, setStockDetail] = useState(activeStock);
  const response = useFetch(endpoint);
  const getStock = symbol =>
    `https://finnhub.io/api/v1/quote?symbol=${symbol}&token=${API_TOKEN}`;

  useEffect(() => {
    setStockDetail(activeStock);

    if (!activeStock) {
      return;
    }

    if (!activeStock.price) {
      setEndpoint(getStock(activeStock.symbol));
    }
  }, [activeStock]);

  useEffect(() => {
    if (response) {
      setStockDetail({
        ...activeStock,
        ...response,
        price: response.c,
        previousPrice: response.pc
      });
      setEndpoint('');
    }
  }, [response]);

  return [stock, setActiveStock];
};

export const useStockNews = category => {
  const [endpoint, setEndpoint] = useState('');
  const [news, setNews] = useState([]);
  const response = useFetch(endpoint);
  const cache = useStorageCache(
    'stock-news',
    { news: response },
    res => !res.news.length
  );

  useEffect(() => {
    const cachedNews = cache && cache.news;

    if (cachedNews && cachedNews.length) {
      setNews(cachedNews);
      const expired = isExpiredNews(cachedNews[0].datetime);

      if (!expired && !category) {
        // Exit early when cache is not expired and no category
        return;
      }
    }

    setEndpoint(getStockNewsEndpoint(category));
  }, [category]);

  useEffect(() => {
    if (response && response.length) {
      setNews(response);
    }
  }, [response]);

  return news;
};

export const useStockChart = (symbol, range) => {
  const [endpoint, setEndpoint] = useState();
  const [updatedResponse, setResponse] = useState(null);
  const response = useFetch(endpoint);

  useEffect(() => {
    const updatedEndpoint = getCandlestickEndpoint(symbol, range);

    setEndpoint(updatedEndpoint);
  }, [symbol, range]);

  useEffect(() => {
    if (response && response.s === 'ok') {
      const data = updateChartData(response, range);

      setResponse(data);
    }
  }, [response]);

  return updatedResponse;
};

export const useStockTicker = (stocks, range = '1D') => {
  const [endpoints, setEndpoints] = useState([]);
  const [updatedResponse, setResponse] = useState([]);
  const response = useFetchAll(endpoints);
  const cache = useStorageCache(
    'stock-charts',
    { charts: updatedResponse },
    res => !res.charts.length
  );

  useEffect(() => {
    const cachedCharts = cache && cache.charts;

    if (cachedCharts && cachedCharts.length) {
      const updated = cachedCharts.map(rehydrateChartData);
      const [head] = updated;
      const expired = isExpired(head.chartData.endDate);

      setResponse(updated);

      if (!expired) {
        // Exit early when cache is not expired
        return;
      }
    }

    // Trigger fetch for candlestick data
    setEndpoints(getCandlestickEndpoints(stocks, range));
  }, [stocks.length]);

  useEffect(() => {
    if (response.length) {
      const updated = response.map((res, index) => {
        if (res.s !== 'ok') {
          return null;
        }

        const stock = stocks[index];
        const chartData = updateChartData(res, range);

        return {
          ...stock,
          chartData
        };
      });

      if (updated[0] !== null) {
        setResponse(updated);
      }
    }
  }, [response]);

  return updatedResponse;
};
