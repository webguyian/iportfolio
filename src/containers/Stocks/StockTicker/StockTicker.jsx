import React from 'react';
import PropTypes from 'prop-types';

import { useStockTicker } from 'containers/Stocks/hooks';
import Stock from 'containers/Stocks/StockTicker/Stock';

const StockTicker = props => {
  const { onClick, stocks } = props;
  const baseClass = 'stock-ticker';

  if (!stocks || !stocks.length) {
    return null;
  }

  const stockData = useStockTicker(stocks);

  if (!stockData || !stockData.length) {
    return null;
  }

  return (
    <div className={baseClass}>
      <ul className={`${baseClass}-list`}>
        {stockData.map(stock => (
          <Stock
            key={stock.displaySymbol}
            onClick={onClick.bind(null, stock)}
            stock={stock}
          />
        ))}
      </ul>
    </div>
  );
};

StockTicker.propTypes = {
  onClick: PropTypes.func.isRequired,
  stocks: PropTypes.array
};

export default StockTicker;
