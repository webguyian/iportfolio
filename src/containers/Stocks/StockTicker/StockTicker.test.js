import React from 'react';
import { create } from 'react-test-renderer';

import { initialStocks } from 'containers/Stocks/constants';
import * as hooks from 'containers/Stocks/hooks';
import StockTicker from './StockTicker';

describe('<StockTicker />', () => {
  // Override prop warning from react-stockcharts
  console.warn = jest.fn();
  hooks.useStockTicker = jest.fn();

  const stocks = [
    {
      c: 320.03,
      h: 320.03,
      l: 319,
      o: 319.365,
      pc: 320.03,
      symbol: 'AAPL',
      displaySymbol: 'AAPL',
      description: 'APPLE INC',
      price: 320.9,
      previousPrice: 320.03,
      chartData: {
        data: [
          {
            date: '2020-02-10T09:00:00.000Z',
            timestamp: 1581325200,
            close: 318
          },
          {
            date: '2020-02-10T09:01:00.000Z',
            timestamp: 1581325260,
            close: 318.25
          },
          {
            date: '2020-02-10T09:04:00.000Z',
            timestamp: 1581325440,
            close: 318.32
          },
          {
            date: '2020-02-10T09:05:00.000Z',
            timestamp: 1581325500,
            close: 318.4
          },
          {
            date: '2020-02-10T09:06:00.000Z',
            timestamp: 1581325560,
            close: 318.4
          }
        ],
        start: 318,
        startDate: '2020-02-10T09:00:00.000Z',
        end: 320.9,
        endDate: '2020-02-11T00:59:00.000Z',
        range: '1D'
      }
    },
    {
      c: 2079,
      h: 2080,
      l: 2077.01,
      o: 2079,
      pc: 2078.08,
      symbol: 'AMZN',
      displaySymbol: 'AMZN',
      description: 'AMAZON.COM INC',
      price: 2140,
      previousPrice: 2079.28,
      chartData: {
        data: [
          {
            date: '2020-02-10T10:40:00.000Z',
            timestamp: 1581331200,
            close: 2082
          },
          {
            date: '2020-02-10T10:48:00.000Z',
            timestamp: 1581331680,
            close: 2081.99
          },
          {
            date: '2020-02-10T11:17:00.000Z',
            timestamp: 1581333420,
            close: 2083
          },
          {
            date: '2020-02-10T11:36:00.000Z',
            timestamp: 1581334560,
            close: 2084.79
          },
          {
            date: '2020-02-10T11:41:00.000Z',
            timestamp: 1581334860,
            close: 2089.85
          }
        ],
        start: 2082,
        startDate: '2020-02-10T10:40:00.000Z',
        end: 2138.8,
        endDate: '2020-02-11T00:53:00.000Z',
        range: '1D'
      }
    }
  ];

  const props = {
    stocks: initialStocks,
    onClick: jest.fn()
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const component = create(<StockTicker {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly without stocks', () => {
    const component = create(<StockTicker {...props} stocks={[]} />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly without stockData', () => {
    hooks.useStockTicker.mockReturnValue([]);

    const component = create(<StockTicker {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly with stockData', () => {
    hooks.useStockTicker.mockReturnValue(stocks);

    const component = create(<StockTicker {...props} />);

    expect(component).toMatchSnapshot();
  });
});
