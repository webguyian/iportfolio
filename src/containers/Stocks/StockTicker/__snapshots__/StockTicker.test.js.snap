// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StockTicker /> renders correctly 1`] = `null`;

exports[`<StockTicker /> renders correctly with stockData 1`] = `
<div
  className="stock-ticker"
>
  <ul
    className="stock-ticker-list"
  >
    <li
      className="stock-tick"
    >
      <button
        className="ui-btn stock-tick-btn"
        onClick={[Function]}
        type="button"
      >
        <div
          className="stock-tick-content"
        >
          <span
            className="ui-text ui-text--body ui-text--bold stock-tick-symbol"
          >
            AAPL
          </span>
          <span
            className="ui-text ui-text--body ui-text--bold stock-tick-price"
          >
            320.90
          </span>
          <span
            className="ui-text ui-text--body stock-tick-difference stock-tick-difference--up"
          >
            +0.27%
          </span>
        </div>
        <div
          className="stock-chart"
        >
          <div
            className="react-stockchart"
            onClick={[Function]}
            style={
              Object {
                "height": 200,
                "position": "relative",
                "width": 345,
              }
            }
          >
            <svg
              className="react-stockchart"
              height={200}
              style={
                Object {
                  "position": "absolute",
                  "zIndex": 6,
                }
              }
              width={345}
            >
              <style
                type="text/css"
              >
                
	.react-stockcharts-grabbing-cursor {
		pointer-events: all;
		cursor: -moz-grabbing;
		cursor: -webkit-grabbing;
		cursor: grabbing;
	}
	.react-stockcharts-crosshair-cursor {
		pointer-events: all;
		cursor: crosshair;
	}
	.react-stockcharts-tooltip-hover {
		pointer-events: all;
		cursor: pointer;
	}
	.react-stockcharts-avoid-interaction {
		pointer-events: none;
	}
	.react-stockcharts-enable-interaction {
		pointer-events: all;
	}
	.react-stockcharts-tooltip {
		pointer-events: all;
		cursor: pointer;
	}
	.react-stockcharts-default-cursor {
		cursor: default;
	}
	.react-stockcharts-move-cursor {
		cursor: move;
	}
	.react-stockcharts-pointer-cursor {
		cursor: pointer;
	}
	.react-stockcharts-ns-resize-cursor {
		cursor: ns-resize;
	}
	.react-stockcharts-ew-resize-cursor {
		cursor: ew-resize;
	}
              </style>
              <defs>
                <clipPath
                  id="chart-area-clip"
                >
                  <rect
                    height={140}
                    width={280}
                    x="0"
                    y="0"
                  />
                </clipPath>
                <clipPath
                  id="chart-area-clip-320.9"
                >
                  <rect
                    height={140}
                    width={280}
                    x="0"
                    y="0"
                  />
                </clipPath>
              </defs>
              <g
                transform="translate(15.5, 10.5)"
              >
                <rect
                  className=""
                  height={140}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onMouseDown={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  onWheel={[Function]}
                  style={
                    Object {
                      "opacity": 0,
                    }
                  }
                  width={280}
                />
                <g
                  className="react-stockcharts-avoid-interaction"
                >
                  <g
                    transform="translate(0, 0)"
                  >
                    <defs>
                      <linearGradient
                        id="gradient-AAPL-no-controls"
                        x1="0"
                        x2="0"
                        y1="100%"
                        y2="0%"
                      >
                        <stop
                          offset="0%"
                          stopColor="#6eb26d"
                          stopOpacity={0.2}
                        />
                        <stop
                          offset="70%"
                          stopColor="#6eb26d"
                          stopOpacity={0.4}
                        />
                        <stop
                          offset="100%"
                          stopColor="#6eb26d"
                          stopOpacity={0.8}
                        />
                      </linearGradient>
                    </defs>
                    <g
                      className="react-stockcharts-area"
                    >
                      <g
                        style={
                          Object {
                            "clipPath": "url(#chart-area-clip-320.9)",
                          }
                        }
                      >
                        <path
                          className="line "
                          d="MNaN,140LNaN,52LNaN,28LNaN,0LNaN,0LNaN,140LNaN,140LNaN,140LNaN,140LNaN,140Z"
                          fill="url(#gradient-AAPL-no-controls)"
                          stroke="none"
                        />
                      </g>
                      <g
                        style={
                          Object {
                            "clipPath": "url(#chart-area-clip-320.9)",
                          }
                        }
                      >
                        <path
                          className="line  "
                          d="MNaN,140LNaN,52LNaN,28LNaN,0LNaN,0"
                          fill="none"
                          stroke="#6eb26d"
                          strokeDasharray="none"
                          strokeOpacity={1}
                          strokeWidth={1}
                        />
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </svg>
          </div>
        </div>
      </button>
    </li>
    <li
      className="stock-tick"
    >
      <button
        className="ui-btn stock-tick-btn"
        onClick={[Function]}
        type="button"
      >
        <div
          className="stock-tick-content"
        >
          <span
            className="ui-text ui-text--body ui-text--bold stock-tick-symbol"
          >
            AMZN
          </span>
          <span
            className="ui-text ui-text--body ui-text--bold stock-tick-price"
          >
            2140.00
          </span>
          <span
            className="ui-text ui-text--body stock-tick-difference stock-tick-difference--up"
          >
            +2.84%
          </span>
        </div>
        <div
          className="stock-chart"
        >
          <div
            className="react-stockchart"
            onClick={[Function]}
            style={
              Object {
                "height": 200,
                "position": "relative",
                "width": 345,
              }
            }
          >
            <svg
              className="react-stockchart"
              height={200}
              style={
                Object {
                  "position": "absolute",
                  "zIndex": 6,
                }
              }
              width={345}
            >
              <style
                type="text/css"
              >
                
	.react-stockcharts-grabbing-cursor {
		pointer-events: all;
		cursor: -moz-grabbing;
		cursor: -webkit-grabbing;
		cursor: grabbing;
	}
	.react-stockcharts-crosshair-cursor {
		pointer-events: all;
		cursor: crosshair;
	}
	.react-stockcharts-tooltip-hover {
		pointer-events: all;
		cursor: pointer;
	}
	.react-stockcharts-avoid-interaction {
		pointer-events: none;
	}
	.react-stockcharts-enable-interaction {
		pointer-events: all;
	}
	.react-stockcharts-tooltip {
		pointer-events: all;
		cursor: pointer;
	}
	.react-stockcharts-default-cursor {
		cursor: default;
	}
	.react-stockcharts-move-cursor {
		cursor: move;
	}
	.react-stockcharts-pointer-cursor {
		cursor: pointer;
	}
	.react-stockcharts-ns-resize-cursor {
		cursor: ns-resize;
	}
	.react-stockcharts-ew-resize-cursor {
		cursor: ew-resize;
	}
              </style>
              <defs>
                <clipPath
                  id="chart-area-clip"
                >
                  <rect
                    height={140}
                    width={280}
                    x="0"
                    y="0"
                  />
                </clipPath>
                <clipPath
                  id="chart-area-clip-2138.8"
                >
                  <rect
                    height={140}
                    width={280}
                    x="0"
                    y="0"
                  />
                </clipPath>
              </defs>
              <g
                transform="translate(15.5, 10.5)"
              >
                <rect
                  className=""
                  height={140}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onMouseDown={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  onWheel={[Function]}
                  style={
                    Object {
                      "opacity": 0,
                    }
                  }
                  width={280}
                />
                <g
                  className="react-stockcharts-avoid-interaction"
                >
                  <g
                    transform="translate(0, 0)"
                  >
                    <defs>
                      <linearGradient
                        id="gradient-AMZN-no-controls"
                        x1="0"
                        x2="0"
                        y1="100%"
                        y2="0%"
                      >
                        <stop
                          offset="0%"
                          stopColor="#6eb26d"
                          stopOpacity={0.2}
                        />
                        <stop
                          offset="70%"
                          stopColor="#6eb26d"
                          stopOpacity={0.4}
                        />
                        <stop
                          offset="100%"
                          stopColor="#6eb26d"
                          stopOpacity={0.8}
                        />
                      </linearGradient>
                    </defs>
                    <g
                      className="react-stockcharts-area"
                    >
                      <g
                        style={
                          Object {
                            "clipPath": "url(#chart-area-clip-2138.8)",
                          }
                        }
                      >
                        <path
                          className="line "
                          d="MNaN,140LNaN,140LNaN,122LNaN,90LNaN,0LNaN,140LNaN,140LNaN,140LNaN,140LNaN,140Z"
                          fill="url(#gradient-AMZN-no-controls)"
                          stroke="none"
                        />
                      </g>
                      <g
                        style={
                          Object {
                            "clipPath": "url(#chart-area-clip-2138.8)",
                          }
                        }
                      >
                        <path
                          className="line  "
                          d="MNaN,140LNaN,140LNaN,122LNaN,90LNaN,0"
                          fill="none"
                          stroke="#6eb26d"
                          strokeDasharray="none"
                          strokeOpacity={1}
                          strokeWidth={1}
                        />
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </svg>
          </div>
        </div>
      </button>
    </li>
  </ul>
</div>
`;

exports[`<StockTicker /> renders correctly without stockData 1`] = `null`;

exports[`<StockTicker /> renders correctly without stocks 1`] = `null`;
